conbench/entities/history.py

```diff
@@ -495,8 +505,9 @@ def _add_rolling_stats_columns_to_df(
     )
 
     # Add column with rolling mean of the means (only inside of the segment)
-    df["rolling_mean_excluding_this_commit"] = (
-        df.groupby(["case_id", "context_id", "hash", "repository", "segment_id"])
+    df.loc[~df.is_outlier, "rolling_mean_excluding_this_commit"] = (
+        df.loc[~df.is_outlier]
```

Comment:
I hate all these `.loc`s, but every other approach I can think of here is also gross. Open to ideas on how to make this prettier.

Reply:
Yeah, pandas isn't pretty. :)

What does this do for rows that are outliers? Are none of these stats columns populated?

Reply:
Right, they'll just be nan. I think that's what we want for plotting purposes—outliers should just get ignored for rolling things. We could locf (forward-fill) it if we want them populated, but I think that's more confusing if anything.